### PR TITLE
gpio driver implementation for the Microchip MEC1501

### PR DESCRIPTION
**Pull request by franciscomunoz**
This is the gpio driver implementation for the Microchip MEC1501.

### CODE REVIEW

**Comment by andrewboie** [comment]
Code looks clean, need to address Kconfig and CODEOWNERS issues

File: drivers/gpio/Kconfig.xec [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Kconfig.xec - Microchip XEC GPIO configuration options
+#
+# Copyright (c) 2019 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig GPIO_XEC
+	bool "XEC Microchip GPIO driver"
+	depends on GPIO
```
Comment (albertofloyd) [inline drivers/gpio/Kconfig.xec]: Let's standardize with depends on SOC_FAMILY_MEC as agreed for i2C driver

File: drivers/gpio/gpio_mchp_xec.c [hunk 2]
```diff
@@ -0,0 +1,505 @@
...
+static int gpio_xec_configure(struct device *dev,
+				int access_op, u32_t pin, int flags)
+{
+	const struct gpio_xec_config *config = dev->config->config_info;
+	__IO u32_t *current_pcr1;
+	u32_t pcr1 = 0;
+	u32_t mask = 0;
+	u32_t gpio_interrupt = 0;
+
+	/* Validate pin number range in terms of curent port */
```
Comment (albertofloyd) [inline drivers/gpio/gpio_mchp_xec.c]: Typo current

File: drivers/gpio/gpio_mchp_xec.c [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+
+static struct gpio_xec_data gpio_xec_port000_036_data;
+
+DEVICE_AND_API_INIT(gpio_xec_port000_036, DT_GPIO_XEC_GPIO000_036_LABEL,
+		gpio_xec_port000_036_init,
+		&gpio_xec_port000_036_data, &gpio_xec_port000_036_config,
+		POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,
+		&gpio_xec_driver_api);
+
+static int gpio_xec_port000_036_init(struct device *dev);
```
Comment (albertofloyd) [inline drivers/gpio/gpio_mchp_xec.c]: Remove duplicate declaration

File: drivers/gpio/gpio_mchp_xec.c (hunk 2 above)
Comment (franciscomunoz) [reply drivers/gpio/gpio_mchp_xec.c]: fixed

File: drivers/gpio/gpio_mchp_xec.c (hunk 3 above)
Comment (franciscomunoz) [reply drivers/gpio/gpio_mchp_xec.c]: removed all the duplicated declarations

File: drivers/gpio/Kconfig.xec (hunk 1 above)
Comment (franciscomunoz) [reply drivers/gpio/Kconfig.xec]: this was also fixed

**Comment by franciscomunoz** [comment]
> Code looks clean, need to address Kconfig and CODEOWNERS issues

PR15802 addresses CODEOWNERS for all the MEC code.

**Comment by andrewboie** [comment]
> > Code looks clean, need to address Kconfig and CODEOWNERS issues
> 
> PR15802 addresses CODEOWNERS for all the MEC code.

You need to remove your -1 from that patch then..

**Comment by franciscomunoz** [comment]
> > > Code looks clean, need to address Kconfig and CODEOWNERS issues
> > 
> > 
> > PR15802 addresses CODEOWNERS for all the MEC code.
> 
> You need to remove your -1 from that patch then..

done

**Review by albertofloyd — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]
can you push a rebase so the tests are repated?

**Comment by franciscomunoz** [comment]
> can you push a rebase so the tests are repated?



> can you push a rebase so the tests are repated?

All checks passed

**Pull request merged**